org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/TracingMessageConsumer.java

```diff
@@ -72,12 +74,14 @@ public TracingMessageConsumer(
 			Map<String, RequestMetadata> sentRequests,
 			Map<String, RequestMetadata> receivedRequests,
 			PrintWriter printWriter,
+			MessageJsonHandler jsonHandler,
```

Comment:
By 

> class needs to fallback to the old behaviour.

do you mean it should create a new Gson instance and try to serialize it without custom adapters? I find this dangerous as it may still throw an exception as before.

Reply:
I think that in general we need to keep the old behavior as a fallback in cases where a `MessageJsonHandler` is unavailable for whatever reason, but make sure that possible exceptions are properly handled and no longer thrown from those `toString` methods. The goal is to avoid any breaking changes in API or behavior.

Reply:
I made another attempt. I didn't like the fact that the jsonHandler could be null on the `MessageTracer`, so I went for a deprecation approach.

Reply:
If you ask me, I liked the previous approach better. To me, the problem looks like a `progressive enhancement` of the old behaviour in cases where a `MessageJsonHandler` is available. As I have mentioned, there might still be cases where it will not be available, so I can see no reason why we should require it in a constructor. This is my 'informed opinion' so to speak. Nevertheless, I really appreciate your effort.

Reply:
Since `MessageTracer` constructor is package private, I take for granted that it is not instantiated/extended outside lsp4j. So changing it is fine.
Regarding `TracingMessageConsumer`, I deprecated the old constructors, but keep them working as before using a fresh `MessageJsonHandler` instance.
If you think the two use cases are valid on the long run, I can remove deprecation tags.
The last "issue" is the removal of `toString` on the various messages. I don't like reverting to the old behavior, since this could be really creating bugs on user side. What about providing `toString` with the best information we could, but not relying on gson serialization?